### ISSUE TITLE
`vulnerability_alerts` set to true does not enable "Dependabot security updates"

### ISSUE THREAD

**Issue by blt** [integrations/terraform-provider-github#588]
### Terraform Version

```
terraform -v
Terraform v0.12.25
+ provider.github v3.1.0
+ provider.google v3.33.0
+ provider.google-beta v3.33.0
+ provider.random v2.3.1
+ provider.template v2.2.0

Your version of Terraform is out of date! The latest version
is 0.13.5. You can update by downloading from https://www.terraform.io/downloads.html
```

### Affected Resource(s)
Please list the resources as a list, for example:
- github_repository

### Terraform Configuration Files
```hcl
resource "github_repository" "repo" {
  name        = var.repo_name
  description = var.repo_description
  visibility  = "private"

  has_issues    = false
  has_projects  = false
  has_wiki      = false
  has_downloads = false

  delete_branch_on_merge = true
  allow_merge_commit     = false
  is_template            = var.template
  vulnerability_alerts   = true

  auto_init      = var.repo_auto_init
  default_branch = var.repo_default_branch 

  dynamic "template" {
    for_each = var.repo_template != "" ? [1] : []
    content {
      repository = var.repo_template
      owner      = "goodwatercap"
    }
  }
}
```

### Expected Behavior

When flagging `vulnerability_alerts` to true we expect the following to be enabled:

* Dependency graph 
* Dependabot alerts
* Dependabot security updates

### Actual Behavior

When flagged `vulnerability_alerts` to true and only the following were enabled: 

* Dependency graph 
* Dependabot alerts

### Steps to Reproduce

1. Flag a `github_repository` with `vulnerability_alerts` to true.
1. `terraform apply`
1. Confirm at https://github.com/ORG/REPO/settings/security_analysis that "Dependabot security udpates" is not enabled. 

### Important Factoids

Nothing unusual. 

### References

None.

**Comment by SanderKnape** [comment]
I just ran into this issue as well, though what I'm seeing is slightly different from what blt is reporting.

Creating a new repository with `vulnerability_alerts: true` will only enable `Dependabot security updates`. The other two options are not enabled.

Running Terraform again will show `vulnerability_alerts   = false -> true`. After applying this, all three options are enabled.

What makes it more interesting is that we enabled these settings on the organization level. So I would expect these settings to be enabled regardless from what I specify in Terraform (see screenshot below).

I'm testing this with private repositories.

![image](https://user-images.githubusercontent.com/12968849/105334288-fd8df500-5bd6-11eb-8f97-6f7af16fd724.png)

**Comment by gionn** [comment]
> What makes it more interesting is that we enabled these settings on the organization level. So I would expect these settings to be enabled regardless from what I specify in Terraform (see screenshot below).

the option states `for new repositories`, so it serve as a default value for new repositories and not an override for the existing ones.

**Comment by SanderKnape** [comment]
Correct. I'm testing this on a newly-created repository through this Terraform provider. So I expect the setting to be enabled.

**Comment by jspiro** [comment]
I am seeing the same. If you re-apply it will correct the bug-induced drift. Not ideal, but at least eventually consistent.

**Comment by kfcampbell** [comment]
I've looked at this a tiny bit and I _believe_ that setting is applied by [this API](https://docs.github.com/en/rest/reference/repos#enable-automated-security-fixes). There's a helper function to set that [vendored into](https://github.com/integrations/terraform-provider-github/blob/master/vendor/github.com/google/go-github/v32/github/repos.go#L600) this project, but it's currently unreferenced. I haven't tested calling that yet. 

Perhaps it'd be appropriate to add this as a new feature with its own syntax, separate from vulnerability_alerts? I wonder how/if that'd conflict with organization settings to enable it by default.

**Comment by will-bluem-olo** [comment]
Hey @kfcampbell - I was poking around this one and it seems there's a missing "Check if automated security fixes are enabled for a repository" API, analogous to this one for [vulnerability alerts](https://docs.github.com/en/rest/repos/repos#check-if-vulnerability-alerts-are-enabled-for-a-repository). Any thoughts on that? I'm looking to contribute here (and also to google/go-github) but hit a wall on the GitHub API. Thanks.

**Comment by kfcampbell** [comment]
@will-bluem-olo that's a great question. The GET 404s, which is too bad. I've asked internally about it and I'll post again here if I learn something useful.

**Comment by kfcampbell** [comment]
Ahh thanks for reminding me! I did not hear anything back, and just bumped the question again.

**Comment by kfcampbell** [comment]
Alright, there's an internal issue created to track this and the team seems receptive. I'm uncertain of the priority but it seems low at this point. :crossed_fingers: :crossed_fingers: :crossed_fingers: they jump on it!

**Comment by bahag-klickst** [comment]
@kfcampbell 
Any news on this

**Comment by GMZwinge** [comment]
With the latest Terraform 1.6.6 and GitHub provider 5.43.0, a `terraform apply -refresh-only` doesn't seem to update the field `vulnerability_alerts` in the .tfstate file with the state in the UI.

**Comment by kfcampbell** [comment]
I wish I had an update, sorry! You might consider asking your GitHub rep (if you're an enterprise customer) or posting [here](https://github.com/orgs/community/discussions/categories/api-and-webhooks) asking for API coverage.

**Comment by thomaslagies** [comment]
Any updates on this so far?

**Comment by lkiii** [comment]
There is a separate resource to enable security. It needs to be combination of `vulnerability_alerts   = true` and 
```
resource "github_repository_dependabot_security_updates" "repo_security" {
  repository = github_repository.repo.id
  enabled    = true
}
```
